core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -596,7 +597,9 @@ private[spark] class TaskSetManager(
     removeRunningTask(tid)
     info.markFailed()
     val index = info.index
-    copiesRunning(index) -= 1
+    if (copiesRunning(index) >= 1) {
```

Comment:
@mccheah do you remember why you added this?  How do we get more task failures than copies running?  If this change was necessary, I wonder whether there's a bug elsewhere.